File: StewardApi/Contracts/Common/AuctionDataEndpoint/AuctionData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

#pragma warning disable SA1600 // ElementsMustBeDocumented (POCO mapped directly from LSP)
#pragma warning disable CS1591 // XML Comments (POCO mapped directly from LSP)
#pragma warning disable SA1516 // Blank Lines (POCO mapped directly from LSP)

namespace Turn10.LiveOps.StewardApi.Contracts.Common.AuctionDataEndpoint
{
    /// <summary>
    ///     Data about a single auction. From LSP.
    /// </summary>
    public class AuctionData
    {
        public Guid AuctionId { get; set; }
        public AuctionReviewState Status { get; set; }
        public ulong SellerXuid { get; set; }
        public Guid SellerProfileId { get; set; }
        public uint OpeningPrice { get; set; }
        public uint BuyoutPrice { get; set; }
        public uint CurrentPrice { get; set; }
        public DateTime CreatedDateUtc { get; set; }
        public DateTime ClosingDateUtc { get; set; }
        public bool IsVipAuction { get; set; }
        public bool IsTurn10Auction { get; set; }
        public DateTime? TimeFlaggedUtc { get; set; }
        public TimeSpan? Duration { get; set; }
        public bool IsFeatured { get; set; }
        public uint AmountInvested { get; set; }
        public bool IsHotDeal { get; set; }
        public int TunerLevel { get; set; }
        public IList<AuctionDataBid> Bids { get; set; }
        public AuctionDataCar Car { get; set; }
        public int UserReportTotal { get; set; }
        public int UserReportCollectionId { get; set; }
        public AuctionReviewState ReportingState { get; set; }
        public bool IsCarRetrieved { get; set; }
        public bool IsPaymentCollected { get; set; }
        public short LiveryLayers { get; set; }
        public short NumberOfLiveryContributors { get; set; }
        public uint NextBidAmount { get; set; }
        public AuctionDataAuctionAction AllowedActions { get; set; }
        public uint TopBidAmount { get; set; }
        public ulong TopBidderXuid { get; set; }
        public int PainterLevel { get; set; }
        public int BidCount { get; set; }
    }
}

File: packages/StewardClient/src/app/shared/models/auction-data.ts
import BigNumber from 'bignumber.js';
import { DateTime, Duration } from 'luxon';
import { GuidLikeString } from './extended-types';

export interface AuctionData {
  auctionId: GuidLikeString;
  status: AuctionDataAuctionStatus;
  sellerXuid: BigNumber;
  sellerProfileId: GuidLikeString;
  openingPrice: BigNumber;
  buyoutPrice: BigNumber;
  currentPrice: BigNumber;
  createdDateUtc: DateTime;
  closingDateUtc: DateTime;
  isVipAuction: boolean;
  isTurn10Auction: boolean;
  timeFlaggedUtc?: DateTime;
  duration: Duration;
  isFeatured: boolean;
  amountInvested: BigNumber;
  isHotDeal: boolean;
  tunerLevel: BigNumber;
  bids: AuctionDataBid[];
  car: AuctionDataCar;
  userReportTotal: BigNumber;
  userReportCollectionId: BigNumber;
  reportingState: AuctionDataReportingState;
  isCarRetrieved: boolean;
  isPaymentCollected: boolean;
  liveryLayers: BigNumber;
  numberOfLiveryContributors: BigNumber;
  nextBidAmount: BigNumber;
  allowedActions: AuctionDataAuctionAction[];
  topBidAmount: BigNumber;
  topBidderXuid: BigNumber;
  paBigNumbererLevel: BigNumber;
  bidCount: BigNumber;
}

export enum AuctionDataAuctionStatus {
  Open = 'Open',
  Cancelled = 'Cancelled',
  Successful = 'Successful',
  Failed = 'Failed',
  Any = 'Any',
}

export enum AuctionDataReportingState {
  Default = 'Default',
  FlaggedForReview = 'FlaggedForReview',
  AdminRemoved = 'AdminRemoved',
  AdminApproved = 'AdminApproved',
}

export enum AuctionDataAuctionAction {
  None = 'None',
  Bid = 'Bid',
  Buyout = 'Buyout',
  ResolveWon = 'ResolveWon',
  ResolveFailedAuction = 'ResolveFailedAuction',
  Restart = 'Restart',
  Cancel = 'Cancel',
  Report = 'Report',
  RemoveBannedItem = 'RemoveBannedItem',
  ReceiveMoneyAuctionOutbid = 'ReceiveMoneyAuctionOutbid',
  ReceiveMoneyAuctionSold = 'ReceiveMoneyAuctionSold',
}

export interface AuctionDataBid {
  xuid: BigNumber;
  profileId: GuidLikeString;
  amount: BigNumber;
  dateUtc: DateTime;
  status: AuctionDataBidStatus;
  isTopBid: boolean;
}

export enum AuctionDataBidStatus {
  Escrow = 'Escrow',
  WaitingForRetrievalByBidder = 'WaitingForRetrievalByBidder',
  WaitingForRetrievalByAuctionOwner = 'WaitingForRetrievalByAuctionOwner',
  Retrieved = 'Retrieved',
  Cancelled = 'Cancelled',
}

export interface AuctionDataCar {
  wasOriginallyFromAuctionHouse: boolean;
  wasOriginallyGifted: boolean;
  speedRating: BigNumber;
  accelerationRating: BigNumber;
  launchRating: BigNumber;
  brakingRating: BigNumber;
  handlingRating: BigNumber;
  versionedLiveryId: GuidLikeString;
  versionedTuneId: GuidLikeString;
  carHistory: AuctionDataCarHistory;
  partsInTrunk: BigNumber[];
  vin: GuidLikeString;
  lastAuctionId: BigNumber;
  carDivisionId: BigNumber;
  isMotorsportEdition: boolean;
  isReward: boolean;
  isOnlineOnly: boolean;
  isUnicorn: boolean;
  powertrainId: BigNumber;
  make: BigNumber;
  carId: BigNumber;
  garageId: BigNumber;
  manColorIndex: BigNumber;
  countryId: BigNumber;
  displacementId: BigNumber;
  aspirationTypeId: BigNumber;
  performanceIndex: BigNumber;
  performanceIndexFriendly: BigNumber;
  carClass: BigNumber;
  year: BigNumber;
  peakPower: BigNumber;
  peakTorque: BigNumber;
  curbWeight: BigNumber;
  weightDistribution: BigNumber;
  isLiveryLocked: boolean;
  numberOfLiveryContributors: BigNumber;
}

export interface AuctionDataCarHistory {
  CurOwnerWinnings: BigNumber;
  CurOwnerNumRaces: BigNumber;
  HighestSkillScore: BigNumber;
  NumSkillPointsEarned: BigNumber;
  NumTimesSold: BigNumber;
  NumOwners: BigNumber;
  CurOwnerPurchaseDate: BigNumber;
  NumRaces: BigNumber;
  NumPodiums: BigNumber;
  TotalRepairs: BigNumber;
  TotalWinnings: BigNumber;
  TimeDriven: BigNumber;
  DistanceDriven: BigNumber;
  Xp: BigNumber;
  NumVictories: BigNumber;
  OriginalOwner: string;
}
